client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -58,46 +61,95 @@ public ReaderGroupConfigBuilder disableAutomaticCheckpoints() {
         * Add a stream and its associated start {@link StreamCut} and end {@link StreamCut} to be read by the
         * readers of a ReaderGroup.
         *
-        * @param scopedStreamName Scoped Name of the Stream.
-        * @param startStreamCut Start {@link StreamCut}
-        * @param endStreamCut End {@link StreamCut}
+        * @param scope Scope of the stream, if it is not specified then {@link ReaderGroup#getScope()} is used for the scope.
+        * @param streamName Stream name.
+        * @param startStreamCut Start {@link StreamCut}.
+        * @param endStreamCut End {@link StreamCut}.
         * @return Reader group config builder.
         */
-       public ReaderGroupConfigBuilder stream(final String scopedStreamName, final StreamCut startStreamCut, final StreamCut endStreamCut) {
-           final Stream stream = Stream.of(scopedStreamName);
+       public ReaderGroupConfigBuilder stream(final String scope, final String streamName,
+                                              final StreamCut startStreamCut, final StreamCut endStreamCut) {
+
+           Exceptions.checkNotNullOrEmpty(streamName, "streamName");
+
+           final String scopedStreamName;
+           if (StringUtils.isBlank(scope)) {
```

Comment:
Why isBlank()? This seems like it's asking for trouble. What if someone has a scope that is just whitespace or ""?

Reply:
That's a good question, but is this something we want to support? Either way, we need to be consistent all around. Please make sure that the naming is consistent across all of our API. In fact, that's one reason we have `NameUtils` under `shared`.

Reply:
Updated the code to use NameUtils